src/shell.c

```diff
@@ -903,6 +902,195 @@ void ShellExecuteCommand(char* p, bool* pbExit)
 	{
 		KePrintMemoryMapInfo();
 	}
+	else if (strcmp (token, "cp") == 0)
+	{
+		LogMsg("Starting to copy...");
+		char* fileNameIn = Tokenize (&state, NULL, " ");
+		if (!fileNameIn)

```

Comment:
One more thing. Consider doing an OR here, like, `!fileName || *fileName == 0`. I should have done that myself.